lib/pages/home/home_page.dart

```diff
@@ -1,26 +1,76 @@
+import 'package:appwrite/appwrite.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+import 'package:no_signal/api/auth/authentication.dart';
 import 'package:no_signal/pages/home/users_list_page.dart';
+import 'package:no_signal/pages/login/login_page.dart';
 import 'package:no_signal/providers/auth.dart';
 import 'package:no_signal/providers/user_data.dart';
 import 'package:no_signal/themes.dart';
 
 import '../../models/popup.dart';
 import '../settings/settings.dart';
 
-class HomePage extends ConsumerWidget {
+class HomePage extends ConsumerStatefulWidget {
```

Comment:
Any specific reason to convert into `ConsumerStatefulWidget` ?

Reply:
If you look I added two functions. One to log out the user and one to show error dialog. I didn't want to make the `Widget Build` function look cluttered. So I moved the functions. You will also notice I moved the `AuthProvider` to the top too.

Reply:
I can revert tho 😂😂 I don't have much experience with `Riverpod`. I usually use `Bloc`. However, your project intrigued me. You used it simply and efficiently. I got tempted to start using it more.

Reply:
I assumed it's better to use the `ConsumerStatefulWidget` in this case due to the additional functions I added. Now coming to think of it, that part is not accessing `dispose` or `init`. I revert?

Reply:
`ConsumerStatefulWidgets` are used when you wanted to use both `ref` and `setState` together. Or maybe you want to use some `controllers` and wanted to dispose them as soon the widget is disposed. It's fine to keep those things inside the build method, they won't make any performance or rebuild issues.

Reply:
Okay 👍🏾

Reply:
But I'll have to keep it in order to access the `mounted` property which is only accessed by a stateful widget which in this case is the `ConsumerStatefulWidget`

Reply:
Great Catch. Let's keep it to `ConsumerStatefulWidget`